### PR TITLE
Added group members kicker

### PR DESCRIPTION
**Pull request by Revadike**
Based on my userscript: https://greasyfork.org/en/scripts/26242-steam-community-group-members-kicker

This is useful for when you want to delete a steam group you made and you have too many members to kick one by one. It kicks everyone, but you. And when you're the only member of the group, the "Leave group" link turns into "Delete group" instead now. 

image
image
image

### CODE REVIEW

**Comment by MxtOUT** [comment]
Not sure how useful this is...

**Comment by Revadike** [comment]
> Not sure how useful this is...

Check OP. The userscript was actually originally made on a request from somebody.

**Comment by MxtOUT** [comment]
What OP?

**Comment by Revadike** [comment]
> This is useful for when you want to delete a steam group you made and you have too many members to kick one by one. It kicks everyone, but you. And when you're the only member of the group, the "Leave group" link turns into "Delete group" instead now.



> What OP?

**Comment by MxtOUT** [comment]
Oh, ok, I thought you were referring to the GreasyFork link.

Anyway, I think there's a reason why Valve hasn't implemented such function yet

**Comment by Revadike** [comment]
> I think there's a reason why Valve hasn't implemented such function yet

Cuz they are lazy? And cuz not everyone has a (large) steam group?

File: js/content/community.js [hunk 1]
```diff
@@ -3459,6 +3459,95 @@ class WorkshopPageClass {
...
+                        window.dialog.Dismiss();
+                    }
+
+                    window.dialog = ShowBlockingWaitDialog("${Localization.str.kick_all_members}", \`${Localization.str.kick_all_loading.replace("__i__", ++i).replace("__total__", total - 1)}\`);
+                }`)
+            }
+
+            function kickMember(steamid) {
+                return new Promise(function(resolve) {
+                    let formData = new FormData();
```
Comment (KarlCastle) [inline js/content/community.js]: Why is this a `new Promise()`? why not return the `RequestData` promise?

File: js/content/community.js [hunk 2]
```diff
@@ -3459,6 +3459,95 @@ class WorkshopPageClass {
...
+                    formData.append("sessionID", HTMLParser.getVariableFromDom("g_sessionID", "string"));
+                    formData.append("action", "kick");
+                    formData.append("memberId", steamid);
+                    formData.append("queryString", "");
+
+                    RequestData.post(window.location.href, formData, {
+                        withCredentials: true
```
Comment (KarlCastle) [inline js/content/community.js]: ... is this route not rate-limited by Valve?

**Comment by KarlCastle** [comment]
> > I think there's a reason why Valve hasn't implemented such function yet
> 
> Cuz they are lazy? And cuz not everyone has a (large) steam group?

Or Valve feels that a (large) steam group is a self-sustaining community that deserves the chance to continue if the creator (owner) no longer wants to be part of it and leaves? Deleting a group also removes its discussion boards and I'm *sure* some of the members that just got kicked are going to be upset that their posts were deleted.

This is only checking if you're an officer? So it's potentially a griefing tool if someone untrustworthy worms their way into kick-access?

**Comment by tfedor** [comment]
Yeah, this is dangerous, irreversible, and has potential for misuse. Closing it.

**Pull request closed without merging**